### PR TITLE
Add tests for OIDC Custom Claims and Custom Resource Quotas

### LINKED ISSUE #16889: Add tests for OIDC Custom Claims and Custom Resource Quotas
#16241 was reopened with the request to add additional tests:

- Add to the existing unit tests an assertion that the fields are enabled
- Add unit test to make sure that the fields addCustomClaims and supportsGroupSearch dont exist for any of the other OIDC providers
- Add e2e test that checks that the custom mapping values are successfully posted to the backend

This issue's purpose is to track these tests so that #16241 can move forward with feature verification.

### PR DESCRIPTION
**Pull request by rak-phillip**
### Summary
This adds additional tests to increase coverage for the custom claims and custom resource quotas features.

Fixes #16889

### Occurred changes and/or fixed issues

- Add e2e tests for custom resource quotas
- Update unit tests for OIDC
- Write e2e tests for OIDC custom claims

### Technical notes summary

#16241 was reopened with the request to add additional tests:

- Add to the existing unit tests an assertion that the fields are enabled
- Add unit test to make sure that the fields addCustomClaims and supportsGroupSearch dont exist for any of the other OIDC providers
- Add e2e test that checks that the custom mapping values are successfully posted to the backend

I'm also including additional e2e tests to assert that the custom resource quotas save data properly.

### Areas or cases that should be tested

e2e & CI should pass.

### Areas which could experience regressions

NA. Adding additional tests to increase coverage.

### Screenshot/Video

NA

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes
- [x] The PR has been reviewed in terms of Accessibility
- [x] The PR has considered, and if applicable tested with, the three Global Roles `Admin`, `Standard User` and `User Base`

### CODE REVIEW

**Comment by rak-phillip** [comment]
@IsaSih I'm adding you as a reviewer as well so that you can confirm the updated tests provide sufficient coverage.

File: shell/edit/auth/__tests__/oidc.test.ts [hunk 1]
```diff
@@ -297,5 +301,18 @@ describe('oidc.vue', () => {
       expect(groupsClaim.exists()).toBe(false);
       expect(emailClaim.exists()).toBe(false);
     });
+
+    it.each([
+      ['cognito'],
```
Comment (mantis-toboggan-md) [inline shell/edit/auth/__tests__/oidc.test.ts]: Just out of curiosity why `each` with only one option? Not a blocker, this test and the others make sense to me and are running fine.

**Review by mantis-toboggan-md — DISMISSED** [review]

File: shell/edit/auth/__tests__/oidc.test.ts (hunk 1 above)
Comment (IsaSih) [inline shell/edit/auth/__tests__/oidc.test.ts]: I'd add more auth providers to this list or simplifying it to a standard `it` block

File: shell/edit/auth/__tests__/oidc.test.ts [hunk 2]
```diff
@@ -297,5 +301,18 @@ describe('oidc.vue', () => {
...
+    ])('addCustomClaims and supportsGroupSearch fields do not exist for provider "%s"', async(providerId: string) => {
+      wrapper.setData({ model: { ...mockModel, id: providerId } });
+      await nextTick();
+
+      const addCustomClaimsCheckbox = wrapper.find('[data-testid="input-add-custom-claims"]');
+      const groupSearchCheckbox = wrapper.find('[data-testid="input-group-search"]');
+
+      expect(addCustomClaimsCheckbox.exists()).toBe(false);
+      expect(groupSearchCheckbox.exists()).toBe(false);
+    });
```
Comment (IsaSih) [inline shell/edit/auth/__tests__/oidc.test.ts:331]: I'd add one more test case, as I noticed that we can check the enableCustomClaims, but the input fields are not mandatory. So, I'd also verify this case when the checkbox is selected, but the fields are undefined.

**Review by IsaSih — CHANGES_REQUESTED** [review]

File: cypress/e2e/tests/pages/explorer2/project-namespace.spec.ts [hunk 3]
```diff
@@ -74,6 +74,87 @@ describe('Projects/Namespaces', { tags: ['@explorer2', '@adminUser'] }, () => {
...
+      createProjectPage.inputCustomType().expectToBeDisabled();
+    });
+
+    it('identifier input is enabled when the Custom resource type is selected', () => {
+      projectsNamespacesPage.baseResourceList().masthead().create();
+      createProjectPage.resourceDetail().createEditView().nameNsDescription()
+        .name()
+        .set('test-1234');
+      createProjectPage.tabResourceQuotas().click();
+      createProjectPage.btnAddResource().click();
```
Comment (IsaSih) [inline cypress/e2e/tests/pages/explorer2/project-namespace.spec.ts:95]: I'd call the `selectResourceTypeByLabel('Custom')` here, as we might add more resource types in the future that could make the custom type not the one that is already populated as a new row is added. Same logic to be applied for the other tests

File: cypress/e2e/tests/pages/explorer2/project-namespace.spec.ts [hunk 4]
```diff
@@ -74,6 +74,87 @@ describe('Projects/Namespaces', { tags: ['@explorer2', '@adminUser'] }, () => {
...
+          createProjectPage.inputNamespaceDefaultLimit().set('2');
+          createProjectPage.resourceDetail().createEditView().create();
+
+          cy.wait('@createProjectRequest').then(({ request }) => {
+            expect(request.body.resourceQuota.limit.extended['example.io/custom-resource']).to.equal('5');
+            expect(request.body.namespaceDefaultResourceQuota.limit.extended['example.io/custom-resource']).to.equal('2');
+          });
+        });
+      });
+
```
Comment (IsaSih) [inline cypress/e2e/tests/pages/explorer2/project-namespace.spec.ts:150]: I'd like to request to also add test that cover trying to post duplicate custom resource quotas, and try to post empty resource quotas (just providing the identifier, but no quotas

**Review by IsaSih — CHANGES_REQUESTED** [review]

File: shell/edit/auth/__tests__/oidc.test.ts (hunk 1 above)
Comment (rak-phillip) [reply shell/edit/auth/__tests__/oidc.test.ts]: I think that I had refactored from a previous implementation. I've removed the `each` specification for now.

Comment (rak-phillip) [reply shell/edit/auth/__tests__/oidc.test.ts]: I've removed the each specification for now.

https://github.com/rancher/dashboard/pull/16740#discussion_r2874012920

**Comment by rak-phillip** [comment]
@IsaSih this is ready for another review
